### PR TITLE
Allow for different calypso roster

### PR DESCRIPTION
**Pull request by ineiti**
Before: Calypso roster was always the same as the byzcoin roster
Now: Allow for different roster

### CODE REVIEW

File: external/js/cothority/src/calypso/calypso-rpc.ts [hunk 1]
```diff
@@ -17,11 +17,15 @@ import { DecodeKey, OnChainSecretInstance } from "./calypso-instance";
 export class OnChainSecretRPC {
     static serviceID = "Calypso";
     private socket: IConnection;
-    private list: ServerIdentity[];
+    private readonly list: ServerIdentity[];
```
Comment (Gilthoniel) [inline external/js/cothority/src/calypso/calypso-rpc.ts]: Why is this necessary ?

Comment (ineiti) [reply external/js/cothority/src/calypso/calypso-rpc.ts]: you mean it should be only `private`, but not `private readonly`? I think it was an IntelliJ suggestion, because the variable is not changed outside of the constructor. So I accepted the suggestion.
